include/xtensor/xbuilder.hpp

```diff
@@ -267,6 +267,84 @@ namespace xt
             }
         };
 
+        // These methods should be private methods of linspace_generator, however thi leads
+        // to ICE on VS2015
+        template <class R, class E, class U, class X, XTL_REQUIRES(xtl::is_integral<X>)>
+        inline void linspace_assign_to(xexpression<E>& e, U start, U, X step, bool) noexcept
+        {
+            arange_assign_to(e, start, step);
+        }
+
+        template <class R, class E, class U, class X, XTL_REQUIRES(xtl::negation<xtl::is_integral<X>>)>
+        inline void linspace_assign_to(xexpression<E>& e, U start, U stop, X step, bool endpoint) noexcept
+        {
+            auto& buf = e.derived_cast().storage();
+            using size_type = decltype(buf.size());
+            using mult_type = get_mult_type<U, X>;
+            size_type num = buf.size();
+            for(size_type i = 0; i < num; ++i)
+            {
+                buf[i] = static_cast<R>(start + step * mult_type(i));
+            }
+            if (num > 0 && endpoint)
+            {
+                buf[num-1] = static_cast<R>(stop);
+            }
+        }
+
+        template <class T, class R = T, class S = T>
+        class linspace_generator
+        {
+        public:
+
+            using value_type = R;
+            using step_type = S;
+
+            linspace_generator(T start, T stop, S step, bool endpoint)
+                    : m_start(start), m_stop(stop), m_step(step), m_endpoint(endpoint)
+            {
+            }
+
+            template <class... Args>
+            inline R operator()(Args... args) const
+            {
+                return access_impl(args...);
+            }
+
+            template <class It>
+            inline R element(It first, It) const
+            {
+                // Avoids warning when T = char (because char + char => int!)
+                using mult_type = get_mult_type<T, S>;
+                return static_cast<R>(m_start + m_step * mult_type(*first));
```

Comment:
I guess we need to fix this line as well.

Reply:
I think this should forward to `access_impl` to avoid code duplication (as it is the case now).

Reply:
Done!